pkg/config/piped.go

```diff
@@ -947,7 +947,7 @@ func (n *NotificationReceiverSlack) Validate() error {
 	if n.HookURL != "" {
 		return nil
 	}
-	if n.OAuthToken == "" || n.ChannelID == "" {
+	if (n.OAuthToken != "" && n.ChannelID == "") || (n.OAuthToken == "" && n.ChannelID != "") {
```

Comment:
Not sure why we should adopt this over the previous version 😄 Previous one fit the error message more than the current condition and it's easier to read 🤔

Reply:
The previous version returns error, if HookURL and OauthToken, ChannelID doesn't set.

Reply:
wdyt?

Reply:
> The previous version returns error, if HookURL and OauthToken, ChannelID doesn't set.

I don't think so, after L947 ( If `n.HookURL != ""` check ), the HookURL must be empty. Then after that, `if n.OAuthToken == "" || n.ChannelID == ""` if one of oauthToken or channelID is empty, it returns error. That's what I mean, isn't it?

Reply:
In the first place, NotificationReceiverSlack requires HookURL or OAuthToken and ChannelID. So it's no way that both parameters is empty. I completely misunderstand this point. Sorry 🐱